legend-engine-test-runner-mapping/src/main/java/org/finos/legend/engine/testable/mapping/extension/MappingTestRunner.java

```diff
@@ -0,0 +1,241 @@
+                    testResult.testable = getElementFullPath(pureMapping, pureModel.getExecutionSupport());
+                    testResult.atomicTestId.testSuiteId = suite.id;
+                    results.add(testResult);
+                }
+            }
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException("Exception occurred executing service test suites.\n" + e);
+        }
+        finally
+        {
+            List<Closeable> runtimeCloseables = connections.stream().map(x -> x.getTwo()).flatMap(Collection::stream).collect(Collectors.toList());
+            if (runtimeCloseables != null)
+            {
+                runtimeCloseables.stream().forEach(closeable ->
+                {
+                    try
+                    {
+                        closeable.close();
+                    }
+                    catch (IOException e)
+                    {
+                        LOGGER.warn("Exception occurred closing closeable resource" + e);
+                    }
+                });
+            }
+        }
+        return results;
+    }
+
+    private List<Pair<Connection, List<Closeable>>> buildTestConnections(PureModelContextData pmcd, List<StoreTestData> storeTestData)
+    {
+        List<Pair<String, EmbeddedData>> connectionInfo = storeTestData.stream().map(testData ->
+        {
+            EmbeddedData embeddedData = null;
+            if (testData.data instanceof DataElementReference)
+            {
+                DataElement dataElement = Iterate.detect(pmcd.getElementsOfType(DataElement.class), e -> ((DataElementReference) testData.data).dataElement.equals(e.getPath()));
+                embeddedData = dataElement.data;
+            }
+            else
+            {
+                embeddedData = testData.data;
+            }
+            return Tuples.pair(testData.store, embeddedData);
+        }).collect(Collectors.toList());
+
+        List<Pair<Connection, List<Closeable>>> connections = connectionInfo.stream().map(pair -> this.getTestConnectionFromFactories(pair.getOne(), pair.getTwo(), pmcd)).collect(Collectors.toList());
+        return connections;
+    }
+
+    private TestResult executeMappingTest(MappingTest test, Mapping pureMapping, PureModel pureModel, MutableList<PlanTransformer> planTransformers, RichIterable<? extends Root_meta_pure_extension_Extension> routerExtensions)
+    {
+        AtomicTestId atomicTestId = new AtomicTestId();
+        atomicTestId.atomicTestId = test.id;
+        try
+        {
+            LambdaFunction<?> pureLambda = HelperValueSpecificationBuilder.buildLambda(test.query, new CompileContext.Builder(pureModel).build());
+            SingleExecutionPlan executionPlan = PlanGenerator.generateExecutionPlan(pureLambda, pureMapping, this.runtime, null, pureModel,this.pureVersion, PlanPlatform.JAVA, null, routerExtensions, planTransformers);
+            Result result = this.executor.execute(executionPlan);
```

Comment:
Are we using deprecated function here?

Reply:
changes it to the newer non-deprecated version